tap-snapshots/test-arborist-build-ideal-tree.js-TAP.test.js

```diff
@@ -29131,6 +29131,77 @@ Node {
+Node {
+  "children": Map {
+    "@isaacs/testing-file-transitive-a" => Node {
+      "edgesIn": Set {
+        Edge {
+          "from": "",
+          "name": "@isaacs/testing-file-transitive-a",
+          "spec": "file:testing-file-transitive-a/isaacs-testing-file-transitive-a-1.0.0.tgz",
+          "type": "prod",
+        },
+      },
+      "edgesOut": Map {
+        "@isaacs/testing-file-transitive-b" => Edge {
+          "name": "@isaacs/testing-file-transitive-b",
+          "spec": "file:../testing-file-transitive-b/isaacs-testing-file-transitive-b-1.0.0.tgz",
+          "to": "node_modules/@isaacs/testing-file-transitive-b",
+          "type": "prod",
+        },
+        "@isaacs/testing-file-transitive-c" => Edge {
+          "name": "@isaacs/testing-file-transitive-c",
+          "spec": "file:./c/isaacs-testing-file-transitive-c-1.0.0.tgz",
+          "to": "node_modules/@isaacs/testing-file-transitive-c",
+          "type": "prod",
+        },
+      },
+      "location": "node_modules/@isaacs/testing-file-transitive-a",
+      "name": "@isaacs/testing-file-transitive-a",
+      "resolved": "file:{CWD}/test/fixtures/tarball-dependencies/testing-file-transitive-a/isaacs-testing-file-transitive-a-1.0.0.tgz",
+    },
+    "@isaacs/testing-file-transitive-b" => Node {
+      "edgesIn": Set {
+        Edge {
+          "from": "node_modules/@isaacs/testing-file-transitive-a",
+          "name": "@isaacs/testing-file-transitive-b",
+          "spec": "file:../testing-file-transitive-b/isaacs-testing-file-transitive-b-1.0.0.tgz",
+          "type": "prod",
+        },
+      },
+      "location": "node_modules/@isaacs/testing-file-transitive-b",
+      "name": "@isaacs/testing-file-transitive-b",
+      "resolved": "file:{CWD}/test/fixtures/tarball-dependencies/testing-file-transitive-b/isaacs-testing-file-transitive-b-1.0.0.tgz",
+    },
+    "@isaacs/testing-file-transitive-c" => Node {
+      "edgesIn": Set {
+        Edge {
+          "from": "node_modules/@isaacs/testing-file-transitive-a",
+          "name": "@isaacs/testing-file-transitive-c",
+          "spec": "file:./c/isaacs-testing-file-transitive-c-1.0.0.tgz",
+          "type": "prod",
+        },
+      },
+      "location": "node_modules/@isaacs/testing-file-transitive-c",
+      "name": "@isaacs/testing-file-transitive-c",
+      "resolved": "file:{CWD}/test/fixtures/tarball-dependencies/testing-file-transitive-a/c/isaacs-testing-file-transitive-c-1.0.0.tgz",
+    },
+  },
+  "edgesOut": Map {
+    "@isaacs/testing-file-transitive-a" => Edge {
+      "name": "@isaacs/testing-file-transitive-a",
+      "spec": "file:testing-file-transitive-a/isaacs-testing-file-transitive-a-1.0.0.tgz",
+      "to": "node_modules/@isaacs/testing-file-transitive-a",
+      "type": "prod",
+    },
+  },
+  "location": "",
+  "name": "tarball-dependencies",
+  "resolved": null,
+}
+`
+
```

Comment:
just curious about the resolved `spec` values that varies across this snapshot, some of them point at `file:testing-file-transitive*` others to `file:../testing-file-transitive*`, some other to `file:./c/`

took a quick look at the fixtures and while the `file:./c/` path seems to make sense since that's a nested folder - I'm quite confused to why we then have `file:testing-file-transitive-a/` and `file:../testing-file-transitive-b/`, it seems to me it would have to be either `file:./` (for transitive-a) and then `file:../testing-file-transitive-b/` or `file:../testing-file-transitive-a/` and `file:../testing-file-transitive-b/`

Reply:
The reason for the `../` there is that the edges are resolved relative to the folder where the tarball is found.

So, the root has:

```
+-- testing-transitive-a/isaacs-testing-transitive-a-1.0.0.tgz
+-- testing-transitive-b/isaacs-testing-transitive-b-1.0.0.tgz
```

The `testing-transitive-a` package's package.json has a dep on `../testing-transitive-b/isaacs-testing-transitive-b-1.0.0.tgz`.  Ie, a reference from the `a` package to the `b` package which is its peer.

When we resolve `file` spec edges on `file`-resolved packages, the resulting target is the resolution against the dirname of the requiring module's resolved value.

Reply:
That is, the root depends on `file:testing-file-transitive-a/isaacs-testing-file-transitive-a-1.0.0.tgz`, and then `a` depends on the file at `./testing-file-transitive-b/isaacs-testing-file-transitive-b-1.0.0.tgz`, which from a's pov is `../testing-file-transitive-b/isaacs-testing-file-transitive-b-1.0.0.tgz`.

Reply:
I see! I think one piece of information I skipped and realized later is that these `spec` values are from edges, so in reality it's just whatever was declared there in this fixture deps 😊